### PR TITLE
chore: add docker

### PR DESCRIPTION
**Pull request by mkxbl**
Add docker:
- ckb dev chain
- ckb indexer
- geth private chain
- docker-compose

### CODE REVIEW

File: docker/ckb/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM nervos/ckb
```
Comment (huwenchao) [inline docker/ckb/Dockerfile:1]: add a tag here to lock the version

File: docker/ckb/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+if ! [ -f ckb.toml ]; then
+  /bin/ckb init --chain "$CKB_CHAIN" --ba-arg "$BA_ARG" --ba-code-hash "$BA_CODE_HASH" --ba-hash-type "$BA_HASH_TYPE" --ba-message "$BA_MESSAGE"
+fi
+
+exec /bin/ckb run &
```
Comment (huwenchao) [inline docker/ckb/entrypoint.sh:7]: Does the logs of `ckb run` redirect to docker log in this way?

File: docker/ckb-indexer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+     /usr/lib/x86_64-linux-gnu/libssl.so.* \
+     /usr/lib/x86_64-linux-gnu/libcrypto.so.* \
+     /usr/lib/x86_64-linux-gnu/
+
+COPY --from=ckb-indexer-builder \
+    /var/lib/ckb-indexer/target/debug/ckb-indexer /bin/
+
+EXPOSE 8116
+
+CMD RUST_LOG=info ckb-indexer -s /tmp/ckb-indexer-test -c http://ckb:8114
```
Comment (huwenchao) [inline docker/ckb-indexer/Dockerfile:21]: `-c http://ckb:8114` only works when this container links to a ckb container named `ckb`.
Your can use a env var here. When we use ckb-indexer standalone, we can pass new env to the container.

ref: <https://vsupalov.com/docker-env-vars/>

File: docker/ckb/Dockerfile (hunk 1 above)
Comment (mkxbl) [reply docker/ckb/Dockerfile:1]: ok

File: docker/ckb-indexer/Dockerfile (hunk 3 above)
Comment (mkxbl) [reply docker/ckb-indexer/Dockerfile:21]: I can just remove `-c http://ckb:8114`, which is the default start command.

If someone wants customized start command, he can overwrite `CMD` with `docker run`

File: docker/ckb/entrypoint.sh (hunk 2 above)
Comment (mkxbl) [reply docker/ckb/entrypoint.sh]: fixed

File: docker/ckb/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+if ! [ -f ckb.toml ]; then
+  /bin/ckb init --chain "$CKB_CHAIN" --ba-arg "$BA_ARG" --ba-code-hash "$BA_CODE_HASH" --ba-hash-type "$BA_HASH_TYPE" --ba-message "$BA_MESSAGE"
+fi
+
+mkdir ckb_log \
+	&& touch ckb_log/output.log ckb_log/error.log \
+	&& ln -sf /dev/stdout ./ckb_log/output.log \
+  && ln -sf /dev/stderr ./ckb_log/error.log
+
+exec /bin/ckb run > ./ckb_log/output.log >> ./ckb_log/error.log &
```
Comment (huwenchao) [inline docker/ckb/entrypoint.sh:12]: Sorry, seems like I'm wrong with it.

```
command_a &
command_b
```

will just put all command_a stdout/stderr and command_b stdout/stderr in the terminal.

You do not need to bother with the log redirect.

Comment (mkxbl) [reply docker/ckb/entrypoint.sh]: updated

**Pull request merged**